src/NLU.DevOps.ModelPerformance/TestCaseSource.cs

```diff
@@ -162,25 +165,9 @@ internal static IEnumerable<TestCase> ToIntentTestCases(LabeledUtterancePair pai
             var expected = expectedUtterance.Intent;
             var actual = actualUtterance.Intent;
 
-            bool isNoneIntent(string intent)
-            {
-                return intent == null || intent == "None";
-            }
-
-            if (isNoneIntent(expected) && isNoneIntent(actual))
+            if (expected == null || actual == null)
             {
-                yield return TrueNegative(
-                    pair.UtteranceId,
-                    ComparisonTargetKind.Intent,
-                    expectedUtterance,
-                    actualUtterance,
-                    score,
-                    null,
-                    new[] { text },
-                    "Both intents are 'None'.",
-                    "Intent");
-
-                yield break;
+                throw new InvalidOperationException("Intent information is missing");
```

Comment:
Curious why we no longer support this?

Reply:
Today's code does not support the compare command for Lex results since the format doesn't supports "intentName" #252  will track the work to support Lex properly.
For LUIS and DF null intents are not allowed. It means there is no "intent" entry in the JSON which makes the comparison void and we should throw an error when this happens.

---
In reply to: [379197665](https://github.com/microsoft/NLU.DevOps/pull/254#discussion_r379197665) [](ancestors = 379197665)

Reply:
I'm confused, because when we use the `test` command with `--service lex`, we map the `intentName` field to `intent` (here: https://github.com/microsoft/NLU.DevOps/blob/master/src/NLU.DevOps.Lex/LexNLUTestClient.cs#L97)

Reply:
oh darn. I didn't run the test command before. I ran the compare command with Lex's input format which had "currentIntent" and took directly the response from lex to do the comparison.
So I should definitely consider the null intent for Lex's now . Maybe I should just create a "null" intent for Lex and calculate the numbers just like we do for None intent for LUIS

Reply:
but we won't have the provider when running compare..

Reply:
Thinking about #256 that you just mentioned, it would solve the problem and actually make the usage of NLU.DevOps clearer per provider especially that with the generic model we lose some data points for comparison

Reply:
Made the change to support null intent and treat it just like None and default fallback.